Sources/HaCWebsiteLib/Events/GeneralEvent.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+
+struct GeneralEvent: PostCardRepresentable {
+    let title : String
+    let time : DateInterval
+    let tagLine : String
+    let color : String
+    let hypePeriod : DateInterval
+    let tags : [String]
+    let eventDescription : Text
+    let imageURL : String? 
+    let location : Location?  //for room names etc.
+    let facebookEventID : Double?
```

Comment:
How come the event is a `Double`? Do you think `String?` would be a better type?

Reply:
I put it as a `Double` because I found somewhere that Facebook uses a 64 bit ID. I guess wrt `Strings` I'm not really sure what the benefit of one or the other would be.

Reply:
I worry about `Double` because we can't really get away with floating-point precision when we're trying to reference an exact thing.

The [Facebook API reference](https://developers.facebook.com/docs/graph-api/reference/event/) says that IDs are a "numeric string", so while they are a number, they come back as a string. We can treat them as black boxes in this case (I believe Facebook expects us to), so if IDs ever switch to something else (like a UUID for example), we won't get hit by it.